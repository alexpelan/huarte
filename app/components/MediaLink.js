import React, {
  StyleSheet,
  Text
} from "react-native";

import MediaWebView from "./MediaWebView";
import styles from "../styles/styles";

// Should this be truly polymorphic? 
const MediaLink = React.createClass({

  linkText: function() {
    if (this.props.media.type === "audio") {
      return "Play audio";
    } else if (this.props.media.type === "image") {
      return "View image";
    } 
  },

  showMedia: function() {
    this.props.navigator.push({
      title: "Media",
      component: MediaWebView,
      passProps: {
        url: this.props.media.url
      }
    })
  },

  render: function() {
    return (
        <Text style={[styles.hyperlink, styles.scoreText]} onPress={() => { this.showMedia()}}>{this.linkText()}</Text>
      );
  }
});

export default MediaLink;