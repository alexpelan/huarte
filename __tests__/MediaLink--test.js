import 'react-native';
import React from 'react';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import MediaLink from '../app/components/MediaLink';

test('audio link', () => {
  const tree = renderer.create(
    <MediaLink media={{ type: 'audio', url: 'https://www.google.com/myFile.mp3' }} />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('image link', () => {
  const tree = renderer.create(
    <MediaLink media={{ type: 'image', url: 'https://www.google.com/myFile.jpg' }} />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
