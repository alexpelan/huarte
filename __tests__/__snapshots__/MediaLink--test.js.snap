// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`audio link 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  disabled={false}
  ellipsizeMode="tail"
  onPress={[Function]}
  style={
    Array [
      Object {
        "textDecorationLine": "underline",
      },
      Object {
        "color": "white",
        "fontSize": 18,
        "textAlign": "center",
      },
    ]
  }
>
  Play audio
</Text>
`;

exports[`image link 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  disabled={false}
  ellipsizeMode="tail"
  onPress={[Function]}
  style={
    Array [
      Object {
        "textDecorationLine": "underline",
      },
      Object {
        "color": "white",
        "fontSize": 18,
        "textAlign": "center",
      },
    ]
  }
>
  View image
</Text>
`;
